metashare/api/jobs.py

```diff
@@ -284,15 +227,16 @@ def commit_changes_from_org(scratch_org, user, desired_changes, commit_message):
         scratch_org.latest_revision_numbers = sf_changes.get_latest_revision_numbers(
             scratch_org
         )
-        scratch_org.save()
-
-    async_to_sync(push_message_about_instance)(
-        scratch_org,
-        {
-            "type": "GITHUB_CHANGES_COMMITTED",
-            "payload": ScratchOrgSerializer(scratch_org).data,
-        },
-    )
+    except Exception as e:
+        async_to_sync(report_scratch_org_error)(
+            scratch_org, e, "SCRATCH_ORG_COMMIT_CHANGES_FAILED"
+        )
+        tb = traceback.format_exc()
+        logger.error(tb)
+        raise
+    finally:
```

Comment:
Hm, using `finally` here (and above in `get_unsaved_changes`) means in an error case we get both the error message (e.g. `SCRATCH_ORG_COMMIT_CHANGES_FAILED`) _and_ the success message (e.g. `GITHUB_CHANGES_COMMITTED`). I'd propose changing it to an `else`, and then adding to the `except` block:

```py
scratch_org.currently_capturing_changes = False
scratch_org.save()
```

Reply:
This is the scenario where I would want `finalize_*` to handle an error message I think.

Reply:
👍 Yeah, I considered that. Maybe also moving `report_scratch_org_error` to the models as well.